mood-analysis.rb

```diff
@@ -28,5 +33,45 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+text.each do |element|
+  date = element[0..4]
+  smiley = analyze_mood(element)
+  puts date + " "+ smiley
+end
+
+def happy_days(entries)
+  happy_entry = 0
+  general_entry = 0
+  #keep track of # of happy days and entries for each entry
+  entries.map do |entry|
+    happy_entry += 1 if analyze_mood(entry) == ":-)"
+    general_entry += 1
+    break if happy_entry == 3
+  end
+  #output "it takes 5 entries for 3 happy days to occur"
+  if happy_entry < 3
+    puts "Sorry, not enough happy days!"
+  else
+    puts "It takes #{general_entry} entries for 3 happy days to occur."
+  end
+end
```

Comment:
This is fine. But would be better if this method returned the smallest piece of information possible. In this case, just the fixnum of how many days it takes to get 3 happy days. If we do that then we have the ability to utilize this method in different ways instead of being tied to this exact output text / use case.